x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorOnlyRegistry.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.operator;
+
+import org.elasticsearch.action.admin.cluster.configuration.AddVotingConfigExclusionsAction;
+import org.elasticsearch.action.admin.cluster.configuration.ClearVotingConfigExclusionsAction;
+import org.elasticsearch.action.admin.cluster.repositories.delete.DeleteRepositoryAction;
+import org.elasticsearch.action.admin.cluster.repositories.delete.DeleteRepositoryRequest;
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsAction;
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.util.set.Sets;
+import org.elasticsearch.license.DeleteLicenseAction;
+import org.elasticsearch.license.PutLicenseAction;
+import org.elasticsearch.transport.TransportRequest;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.xpack.security.transport.filter.IPFilter.IP_FILTER_ENABLED_HTTP_SETTING;
+import static org.elasticsearch.xpack.security.transport.filter.IPFilter.IP_FILTER_ENABLED_SETTING;
+
+public class OperatorOnlyRegistry {
+
+    public static final Set<String> SIMPLE_ACTIONS = Set.of(AddVotingConfigExclusionsAction.NAME,
+        ClearVotingConfigExclusionsAction.NAME,
+        PutLicenseAction.NAME,
+        DeleteLicenseAction.NAME,
+        // Autoscaling does not publish its actions to core, literal strings are needed.
+        "cluster:admin/autoscaling/put_autoscaling_policy",
+        "cluster:admin/autoscaling/delete_autoscaling_policy",
+        "cluster:admin/autoscaling/get_autoscaling_policy",
+        "cluster:admin/autoscaling/get_autoscaling_capacity");
+
+    // This class is a prototype to showcase what it would look like for operator only settings
+    // It may NOT be included in phase 1 delivery. Also this may end up using Enum Property to
+    // mark operator only settings instead of using the list here.
+    public static final Set<String> SIMPLE_SETTINGS = Set.of(IP_FILTER_ENABLED_HTTP_SETTING.getKey(), IP_FILTER_ENABLED_SETTING.getKey(),
+        // TODO: Use literal strings due to dependency. Alternatively we can let each plugin publish names of operator settings
+        "xpack.ml.max_machine_memory_percent", "xpack.ml.max_model_memory_limit");
```

Comment:
Should we remove these now?

Reply:
Yes removed. Even if we include settings for phase 1, I think it is very likely going to be taking the Enum property approach.